runtimes/node-18.0/server.js

```diff
@@ -5,7 +5,7 @@ const { text: parseText, json: parseJson, buffer: parseBuffer, send } = require(
 const USER_CODE_PATH = '/usr/code-start';
 
 const server = micro(async (req, res) => {
-    if (req.headers[`x-open-runtimes-secret`] !== process.env['OPEN_RUNTIMES_SECRET']) {
+    if ((req.headers[`x-open-runtimes-secret`] ?? '') !== (process.env['OPEN_RUNTIMES_SECRET'] ?? '')) {
```

Comment:
uff this is dangerous 👀 

If we accidentally forget to set the secret, anyone can bypass the this check by not passing the `x-open-runtimes-secret` header

Reply:
We can set default value in Dockerfile. That way developer would need to purposely set it to empty string.
I find it valuable to give this option. Let's say you are running public static hosting. It gives value being able to just visit URL and see website, without having to set header (which browsers dont do)

Reply:
```suggestion
     // This will check for the case when secret is undefined 
    if (!process.env['OPEN_RUNTIMES_SECRET']) {
        return send(res, 500, 'Unauthorized. Provide correct "x-open-runtimes-secret" header.');
    }
    // This will check all the other cases
    if (req.headers[`x-open-runtimes-secret`] !== process.env['OPEN_RUNTIMES_SECRET']) {
        return send(res, 500, 'Unauthorized. Provide correct "x-open-runtimes-secret" header.');
    }
```

Also are the 500 errors displayed in the function logs in the console ? If yes, we need to change the error message to not expose the `x-open-runtimes-secret` header. Its something internal to Appwrite and end-developers need not see this message in the UI. It should be logged to sentry though

Reply:
I dont agree with not allowing empty secret. To compare, it's like not allowing guest to execute Appwrite function.
Anyway, Ill implement this error and add a test for it.